### PR TITLE
Implement supported parts of Spn/UpnEndpointIdentity

### PR DESCRIPTION
**Pull request by iamjasonp**
Reopening this since #852 caused a build break in .NET Native

### CODE REVIEW

File: src/System.Private.ServiceModel/src/System/ServiceModel/UpnEndpointIdentity.cs [hunk 1]
```diff
@@ -21,8 +22,23 @@ public UpnEndpointIdentity(string upnName)
         {
             if (upnName == null)
                 throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("upnName");
+#if SUPPORTS_WINDOWSIDENTITY
+            base.Initialize(Claim.CreateUpnClaim(upnName));
+#else 
+            throw ExceptionHelper.PlatformNotSupported("UpnEndpointIdentity is not supported on this platform");
+#endif // SUPPORTS_WINDOWSIDENTITY
```
Comment (iamjasonp) [inline src/System.Private.ServiceModel/src/System/ServiceModel/UpnEndpointIdentity.cs]: This was the line that caused the .NET Native build break. Claim.CreateUpnClaim was put under an #if SUPPORTS_WINDOWSIDENTITY.

**Comment by roncain** [comment]
It looks like this failed in Linux during the build phase due to no SecurityIdentifier.  @iamjasonp do you know what that is about?

**Comment by iamjasonp** [comment]
Investigating... I'm starting to question my own sanity lately with how my builds differ. 
I'm thinking that this is just a rebase or push gone wrong.

**Comment by iamjasonp** [comment]
Builds for me on Linux and Windows as well as TFS. @roncain, would you be able to buddy build?

**Comment by roncain** [comment]
LGTM.
I buddy-built on Linux, Windows, and in TFS.

**Comment by iamjasonp** [comment]
Thanks! Merging...

**Pull request merged**